### ISSUE TITLE
Update main/prod with work on `dev` branch

### ISSUE THREAD

**Issue by bendnorman** [deployment-gap-model-education-fund/deployment-gap-model#370]
It's time to update the `main` branch and our production data! We haven't done this in a while because we made a large change to a foundational dataset (using gridstatus for some ISOs instead of LBNL) and we needed time to update our dashboards in inform downstream users of the change.

Here's how we can do this:

- [x] Make sure the latest [`full-build`](https://github.com/deployment-gap-model-education-fund/deployment-gap-model/actions/workflows/run-full-build.yml) on `dev` passed.
- [x] If so, merge `dev` into `main`
- [x] Then, create and push a new tag called `v2024.MM.DD`. This will kick off the `full-build` action which will run the ETL, tests and publish the data to GCS and BigQuery. There should be a `v2024.MM.DD` directory in the `gs://dgm-outputs` bucket and the tables in `dbcp-dev-350818.data_warehouse` and `dbcp-dev-350818.data_mart` should have a `version: v2024.MM.DD` label. You can triple check the data was actually updated by seeing if the ISO project tables have GS data in them. If you run `SELECT distinct(`source`) FROM `dbcp-dev-350818.data_mart.iso_projects_long_format` LIMIT 1000` query you should see: "proprietary", "gridstatus", "lbnl". 

If the the full build worked on `dev` it should probably work on `main` once `dev` is merged into `main`. Once you confirm the build is passing on `dev` you can push the new tag the day Yujia wants the data updated.
